test/e2e/diff_test.go

```diff
@@ -375,3 +375,30 @@ data:
 
 	require.Containsf(t, out, expectedOutput, "Did not find expected diff output")
 }
+
+func TestDiffRun(t *testing.T) {
+	env := BuildEnv(t)
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, Logger{}}
+	kubectl := Kubectl{t, env.Namespace, Logger{}}
+
+	name := "not-create-configmap-diff-run"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	yaml1 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config
+`
+
+	_, _ = kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name,
```

Comment:
I think we could add another case for the update scenario (with labels) to have more coverage

Reply:
Create app -> update app -> Verify that labels were not added

Reply:
Yeah sure, I will add a test case for this scenario also in a separate PR.

Reply:
Done.